src/fontra/views/editor/visualization-layer-definitions.js

```diff
@@ -171,6 +171,66 @@ registerVisualizationLayerDefinition({
   },
 });
 
+registerVisualizationLayerDefinition({
+  identifier: "fontra.verticalMetrics",
+  name: "Vertical font metrics",
+  selectionMode: "editing",
+  userSwitchable: true,
+  defaultOn: true,
+  zIndex: 100,
+  screenParameters: { strokeWidth: 1 },
+  colors: { strokeColor: "#0004", zoneColor: "#0001" },
+  colorsDarkMode: { strokeColor: "#FFF6", zoneColor: "#FFF1" },
+  draw: (context, positionedGlyph, parameters, model, controller) => {
+    context.strokeStyle = parameters.strokeColor;
+    context.lineWidth = parameters.strokeWidth;
+
+    if (model.fontSourceInstance === undefined) {
+      return;
+    }
+    const verticalMetrics = model.fontSourceInstance.verticalMetrics;
+    // const italicAngle = model.fontSourceInstance.italicAngle
+    //   ? model.fontSourceInstance.italicAngle * -1
+    //   : 0;
+    const glyphWidth = positionedGlyph.glyph.xAdvance
+      ? positionedGlyph.glyph.xAdvance
+      : 0;
+
+    // glyph box
+    let pathBox = new Path2D();
+    pathBox.rect(
+      0,
+      verticalMetrics.descender.value,
```

Comment:
We need to account for the possibility that "ascender" or "descender" or both do not exist, and don't draw the box in that case.

Reply:
Right, this is indeed true for designspace. But since our data structure does not guarantee this, we still need to check.